hubverse_annotator/app.py

```diff
@@ -161,8 +221,42 @@ def main() -> None:
         if smhubt_to_plot.is_empty():
             st.warning("No forecasts available for current selection.")
         else:
-            forecast_chart = create_quantile_forecast_chart(smhubt_to_plot)
-            st.altair_chart(forecast_chart, use_container_width=True)
+            forecast_layers = create_quantile_forecast_chart(smhubt_to_plot)
+            if eh_table is not None:
+                eh_to_plot = (
+                    eh_table.with_columns(
+                        pl.col("date").alias("target_end_date")
+                    )
+                    .filter(
+                        (pl.col("location") == two_num_loc_abbr)
+                        & (pl.col("target") == selected_target)
+                        & (
+                            pl.col("target_end_date").dt.strftime("%Y-%m-%d")
+                            <= selected_ref_date
+                        )
```

Comment:
- Why do you want / need this filter at all? If there is observed data into the forecast period, we'd like to see it (ideally in a different color)
- Converting to a datetime column to a string to compare to a single string feels backwards. Safer to convert the string to a datetime.

```suggestion
                            pl.col("target_end_date") <= selected_ref_date)
```